node/impl/full/eth.go

```diff
@@ -527,7 +527,20 @@ func (a *EthModule) EthGetTransactionReceiptLimited(ctx context.Context, txHash
 		return nil, xerrors.Errorf("failed to convert %s into an Eth Txn: %w", txHash, err)
 	}
 
-	receipt, err := newEthTxReceipt(ctx, tx, msgLookup, a.ChainAPI, a.EthEventHandler)
+	ts, err := a.Chain.GetTipSetFromKey(ctx, msgLookup.TipSet)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to lookup tipset %s when constructing the eth txn receipt: %w", msgLookup.TipSet, err)
+	}
+
+	// The tx is located in the parent tipset
+	parentTs, err := a.Chain.LoadTipSet(ctx, ts.Parents())
+	if err != nil {
+		return nil, xerrors.Errorf("failed to lookup tipset %s when constructing the eth txn receipt: %w", ts.Parents(), err)
+	}
+
+	baseFee := parentTs.Blocks()[0].ParentBaseFee
```

Comment:
So this gives the base fee after executing the parent tipset of ` parentTs` which is the inclusion tipset for `tx` as `msgLookup.TipSet` is the execution tipset and `parentTs` = `Parent(msgLookup.TipSet)`.

This is correct because we want the base fee the user paid for their tx which is the network base fee before the inclusion tipset was mined.

Reply:
yes, but
> base fee before the inclusion tipset was mined
this is the bit I'm not super confident about, but it _seems_ to be correct based on how everything is wired up today